networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py

```diff
@@ -233,12 +236,22 @@ def extend_hosting_port_info(self, context, port_db, hosting_device,
             hosting_info['cidr_exposed'] = ext_dict['cidr_exposed']
             hosting_info['gateway_ip'] = ext_dict['gateway_ip']
         else:
+            router_id = port_db.device_id
+            router = self.l3_plugin.get_router(context, router_id)
+            # skip routers not created by the user -- they will have
+            # empty-string tenant IDs
```

Comment:
I think it does. If the router has a role set (e.g. "Global" or "Logical Global") then we want to skip it -- because the tenant ID for those routers is the empty string, which we don't want to use for SNAT port allocation (i.e. we want unique tenant iDs, which only appear on the user-created routers).